datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -77,11 +81,40 @@ fn array_array(args: &[ArrayRef]) -> Result<ArrayRef> {
     // do not accept 0 arguments.
     if args.is_empty() {
         return Err(DataFusionError::Internal(
-            "array requires at least one argument".to_string(),
+            "Array requires at least one argument".to_string(),
         ));
     }
 
-    let res = match args[0].data_type() {
+    let data_type = args[0].data_type();
+    let res = match data_type {
+        DataType::List(..) => {
+            let arrays =
```

Comment:
@tustvold  can you offer some suggestions on using the arrow-rs API to build list arrays? Is this the best way to use that API?

Reply:
It would perhaps be nicer to use a combination of https://docs.rs/arrow-array/latest/arrow_array/array/struct.GenericListArray.html#method.try_new and https://docs.rs/arrow-select/latest/arrow_select/concat/index.html

MutableArrayData is not the nicest API to use